runner.sh

```diff
@@ -0,0 +1,190 @@
+# Using sar command requires start and end time in format of %H:%M[:%S]. in this case i used %H:%M to generate a more detailed output.
+sys_trace() {
+export SYS_TRACE_FILE=./sys_trace_${START_TIME}.log
+#export SYS_TRACE_START=${START_TIME##*_}
+export SYS_TRACE_START=$(echo ${START_TIME##*_} | cut -c-5)
+sar 1 3 -n ALL -P ALL -r -u ALL >> ${SYS_TRACE_FILE} & >/dev/null 2>&1
+${COMMAND} >/dev/null 2>&1 &
+cat ${SYS_TRACE_FILE} | grep ${SYS_TRACE_START} > tmp && mv tmp ${SYS_TRACE_FILE}
+wait
+}
+
+# For each failed execution create a ‘pcap’ file with the network traffic during the execution.
+net_trace(){
+export NET_TRACE_PCAP=./net_trace_${START_TIME}.pcap
+tcpdump -U -i any -nnN -t -s 0 -c 20 -w ${NET_TRACE_PCAP} >/dev/null 2>&1 &
+${COMMAND} >/dev/null 2>&1 &
+wait
+}
+
+# For each failed execution create log with all of the system calls ran by the command.
+call_trace(){
+export CALL_TRACE_FILE=./call_trace-${SCRIPT_START_TIME}.log
+strace -ttT -o ${CALL_TRACE_FILE} ${COMMAND} >/dev/null 2>&1
+}
+
+# For each failed execution create stdout and stderr log files with the output of the command.
+log_trace(){
+export STDOUT_FILE=./log_trace-${SCRIPT_START_TIME}.out
+export STDERR_FILE=./log_trace-${SCRIPT_START_TIME}.err
+${COMMAND} 2>>${STDERR_FILE} 1>>${STDOUT_FILE}
+}
+
+# Print a summary of the command return codes.
+summary(){
+# Check if return_codes file is not exists- (probably in case that only '--failed-count' option exists without '-c').
+if [[ ! -f ./return_codes.txt ]] ; then
+    touch "./return_codes.txt"
+    echo "No command has been running. please check again your argument options"
+    help
+else
+    # Remove empty lines
+    sed -i '/^$/d' return_codes.txt
+    # Counts occurrences of each line and prints the return code and the count.
+    sort return_codes.txt | uniq -c | sort -r | awk '{print $2,$1}' OFS='\t' > tmp && mv tmp return_codes.txt
+    # Insert to the first line in file the titles.
+    sed -i '1i Exit-Code Count' return_codes.txt
+    printf "_____Summary_____\n"
+    cat return_codes.txt | column -t -o" | "
+    rm -f ./return_codes.txt
+fi
+}
+
+# Make sure that summary will be printed at any case of exit signals.
+trap summary SIGINT SIGQUIT SIGTSTP
+
+# Check if --failed-count is not sent as argument to the script. in case that only -c exists.
+if [[ -z ${FAILED_COUNT} ]] ;then
+        echo "no failed count option configured"
+        while [[ ${EXEC_COUNT} -lt ${COUNT} ]] ;do
+                ${COMMAND} < /dev/null 2>&1
+                export RESULT=$(echo "$?")
+                echo "${RESULT}" >> ./return_codes.txt
+                ((EXEC_COUNT++))
+        done
+else
+
+# Running until getting to the limit of 'count' or 'failed-count' parameters.
+while [[ ${EXEC_COUNT} -lt ${COUNT} ]] && [[ ${EXEC_FAILED} -lt ${FAILED_COUNT} ]] ; do
+    export START_TIME=$(date +"%Y-%m-%d_%H:%M:%S")
+    ${COMMAND} > /dev/null 2>&1
```

Comment:
In addition to the command invocation here, the command will run again for each "*-trace" flag that was specified. It means that the command will run more times than specified with "-c" flag.

Moreover, the trace files that will be created won't reflect what happened when the command was called in this line.

Do you have any suggestions on how to solve it?